guppylang/checker/expr_checker.py

```diff
@@ -990,6 +996,15 @@ def instantiate_poly(node: ast.expr, ty: FunctionType, inst: Inst) -> ast.expr:
     return with_type(ty, node)
 
 
+def check_function_tensors_enabled(node: ast.expr | None = None) -> None:
```

Comment:
nit: I wonder if the two `check_xxx_enabled` should live together in `experimental.py`